plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -654,20 +654,54 @@ public KVMStoragePool createStoragePool(String name, String host, int port, Stri
             } catch (LibvirtException e) {
                 s_logger.error("Failure in attempting to see if an existing storage pool might be using the path of the pool to be created:" + e);
             }
+        }
+
+        List<String> nfsopts = null;
+        if (type == StoragePoolType.NetworkFilesystem) {
+            if (details != null && details.containsKey("nfsopts")) {
+                nfsopts = Arrays.asList(details.get("nfsopts").replaceAll("\\s", "").split(","));
+            }
+
+            if (sp != null && nfsopts != null) {
+                try {
+                    LibvirtStoragePoolDef poolDef = getStoragePoolDef(conn, sp);
+                    Map poolNfsOptsMap = poolDef.getNfsOpts();
+                    boolean optionsDiffer = false;
+                    if (poolNfsOptsMap.size() != nfsopts.size()) {
+                        optionsDiffer = true;
+                    } else {
+                        for (String nfsopt : nfsopts) {
+                            if (!poolNfsOptsMap.containsKey(nfsopt)) {
```

Comment:
what if value changes for a key (say, nfs version updated)?

Reply:
The key here contains both the option and the value. For example the key would be "vers=4.1", so if the version is changing optionsDiffer would be true.

Reply:
Ok, I'm assuming` key=vers` & `value=4.1`